app_dart/lib/src/request_handlers/push_build_status_to_github.dart

```diff
@@ -6,122 +6,140 @@ import 'dart:async';
+      await _insertBigquery(slug, status, branch, log, config);
+      final DatastoreService datastore = datastoreProvider(config.db);
+      await _updatePRs(slug, status, datastore);
+      log.debug('All the PRs for $repo have been updated with $status');
     }
 
     return Body.empty;
   }
 
-  Future<void> _insertBigquery(BuildStatus buildStatus, String branch) async {
-    // Define const variables for [BigQuery] operations.
-    const String projectId = 'flutter-dashboard';
-    const String dataset = 'cocoon';
-    const String table = 'BuildStatus';
-
-    final TabledataResourceApi tabledataResourceApi = await config.createTabledataResourceApi();
-    final List<Map<String, Object>> requestRows = <Map<String, Object>>[];
-
-    requestRows.add(<String, Object>{
-      'json': <String, Object>{
-        'Timestamp': DateTime.now().millisecondsSinceEpoch,
-        'Status': buildStatus.value,
-        'Branch': branch,
-      },
-    });
-
-    // Obtain [rows] to be inserted to [BigQuery].
-    final TableDataInsertAllRequest request = TableDataInsertAllRequest.fromJson(<String, Object>{'rows': requestRows});
+  Future<void> _updatePRs(RepositorySlug slug, String status, DatastoreService datastore) async {
+    final GitHub github = await config.createGitHubClient(slug);
+    final List<GithubBuildStatusUpdate> updates = <GithubBuildStatusUpdate>[];
+    await for (PullRequest pr in github.pullRequests.list(slug)) {
+      final GithubBuildStatusUpdate update = await datastore.queryLastStatusUpdate(slug, pr);
+      if (update.status != status) {
+        log.debug('Updating status of ${slug.fullName}#${pr.number} from ${update.status} to $status');
+        final CreateStatus request = CreateStatus(status);
+        request.targetUrl = 'https://ci.chromium.org/p/flutter/g/engine/console';
+        request.context = 'luci-${slug.name}';
+        if (status != GithubBuildStatusUpdate.statusSuccess) {
+          request.description = config.flutterBuildDescription;
+        }
+        try {
+          await github.repositories.createStatus(slug, pr.head.sha, request);
+          update.status = status;
+          update.updates += 1;
+          update.updateTimeMillis = DateTime.now().millisecondsSinceEpoch;
+          updates.add(update);
+        } catch (error) {
+          log.error('Failed to post status update to ${slug.fullName}#${pr.number}: $error');
+        }
+      }
+    }
+    await datastore.insert(updates);
+  }
 
-    try {
-      await tabledataResourceApi.insertAll(request, projectId, dataset, table);
-    } on ApiRequestError {
-      log.warning('Failed to add build status to BigQuery: $ApiRequestError');
+  /// This function gets called with the last 40 builds fo a given builder ordered
+  /// by creation time starting with the last one first.
+  Future<String> _getLatestStatus(List<LuciTask> tasks) async {
+    for (LuciTask task in tasks) {
+      if (task.ref != 'refs/heads/master') {
```

Comment:
Same here. This omits other branches data.

Reply:
Looking at the code we may actually want to remove the complexity associated to branches I don't see any current use cases for it.

Reply:
I'll wait for @christopherfujino reply, if there are release use cases for tree status in branches I'll add relevant tests.

Reply:
yeah, deleting this is fine

Reply:
This much simplifies the logic!